### DISCUSSION TITLE
Running Oqtane.Server from command line is Erroring

### DISCUSSION THREAD

**Discussion by RodDaSilvaWCO** [oqtane/oqtane.framework#2448] [General]
Hi All,

I am trying to do some Oqtane module development and want to run Oqtane.Server from the command line instead of debugging from VS.  

All I am trying to do is open a command line and proceed to the **~\oqtane.framework-dev\Oqtane.Server\bin\Debug\net6.0** folder and run **Oqtane.Server.exe**.  But I keep getting:

![image](https://user-images.githubusercontent.com/78995177/193286411-8bea6b53-f424-4fb8-b452-b15725d1c47b.png)

The issue is that **webRootPath** is empty at the following line in **InstallationManager**:

![image](https://user-images.githubusercontent.com/78995177/193288285-8e300ded-5c5c-4db7-8830-65b2ca384dfb.png)

Any ideas why this variable is set properly when I hit f5 from visual studio, but not when I run directly from the command line?

Thanks,
Rod

**Comment by sbwalker** [comment]
Oqtane is a web application. The Startup.s class injects IWebHostEnvironment which provides essential information about the physical location of the web root and content folders defined for the web hosting environment. These folder locations are utilized by a variety of foundational capabilities which are executed during the startup process. Running Oqtane.Server.exe from the command line has no web hosting context so these environment attributes will be blank. 

```
        public Startup(IWebHostEnvironment env, ILocalizationManager localizationManager)
```

**Reply by RodDaSilvaWCO** [reply]
<< Running Oqtane.Server.exe from the command line has no web hosting context so these environment attributes will be blank.

You HAVE TO be able to run the Oqtane.Server.exe directly.  Otherwise how would you install it on a Web Server that does not have Visual Studio?  See my comment regarding Publishing below.

**Comment by W6HBR** [comment]
To run from the command line, you need to launch the project via "dotnet run" instead of from the exe.
Below is a batch file I use for doing this:

@ECHO OFF
C:
CD C:\Apps\Oqtane-W6HBR\Oqtane.Framework\Oqtane.Server
dotnet run Oqtane.Server.csproj

**Reply by RodDaSilvaWCO** [reply]
This works in that it allows you to run the Oqtane.Server without VS.  Thanks.  I suppose if I build a Release build for example and then "Publish" it form Visual Studio (or the dotnet command line) then I would be able to run the Oqtane.Sever.exe directly as it should have all the necessary wwwRoot sub folder contents, etc., need to run correctly.

**Reply by sbwalker** [reply]
@RodDaSilvaWCO You may want to look in the Oqtane.Package folder - it contains the command files used to create official Oqtane releases.

**Comment by RodDaSilvaWCO** [comment]
Will do thanks.  However, @BCS-JonW 's suggestion satisfies my immediate need to run Oqtane.Server on my dev machine without VS loading it up.
